client/src/js/components/bhDepotSelectSearch/bhDepotSelect.js

```diff
@@ -0,0 +1,97 @@
+
+angular.module('bhima.components')
+  .component('bhDepotSearchSelect', {
+    templateUrl : 'js/components/bhDepotSelectSearch/bhDepotSelectSearch.html',
+    controller  : DepotSearchSelectController,
+    bindings    : {
+      depotsUuids     : '<',
+      label           : '@?',
+      id              : '@?',
+      onChange        : '&',
+      formName        : '@?',
+    },
+  });
+
+DepotSearchSelectController.$inject = [
+  'DepotService', 'uuid',
+];
+
+/**
+* Multiple Depot Selection Component
+*
+*/
+function DepotSearchSelectController(Depots, uuidService) {
+  const $ctrl = this;
+
+  $ctrl.$onInit = () => {
+  // label to display
+    $ctrl.label = $ctrl.label || 'STOCK.DEPOT';
+    // default for form name
+    $ctrl.formName = $ctrl.formName || 'DepotForm';
+    //
+    $ctrl.depotsSected = $ctrl.depotsSected || [];
+    // init the model
+    $ctrl.depotsUuids = $ctrl.depotsUuids || [];
+
+    $ctrl.componentId = $ctrl.id || uuidService().replace('-', '');
+    $ctrl.depots = [];
+
+    // on search text change
+    setTimeout(() => {
```

Comment:
However, you actually don't need any of this code - uib-typeahead will do all this for you.  All you need to do is pass a function that returns an `$http` request to the `uib-typeahead` directive.  We do this is in [bhFindPatient](https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/modules/templates/bhFindPatient.tmpl.html#L46).

It looks like this:
```html
 uib-typeahead="patient as patient.display_name for patient in $ctrl.searchByName($viewValue)"
```
where `$ctrl.searchByName` is [this function](https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/js/components/bhFindPatient.js#L165):
```js
  function searchByName(text) {
    $ctrl.loadStatus = 'loading';

    // format query string parameters
    const options = {
      display_name : text.toLowerCase(),
      limit        : LIMIT,
    };

    return Patients.searchByName(options);
};
```

This means you can just set `loadDepots()` as the function to call with `$viewValue` and it will then load the required depots.

Reply:
thanks